lib/v2/screens/profile_screens/profile/interactor/mappers/upgrade_citizenship_result_mapper.dart

```diff
@@ -14,10 +15,14 @@ class UpgradeCitizenshipResultMapper extends StateMapper {
       );
     } else {
       /// Show citizenship upgrade success
+      ProfileModel? profile = results[0].valueOrNull;
```

Comment:
you could just directly change the profile state to the new state

loading the profile here has additional chances of errors, need to test for error results, need to make a bunch of calls which we don't need

If the result there - which is the result of makeresident or makecitizen - is a success, then the profile has the new state for sure.